### PR TITLE
LPS-175677 | Test Automation | Master

### PR DESCRIPTION
**Pull request by GABRIELCMSANTOS**
https://issues.liferay.com/browse/LPS-175677

### CODE REVIEW

**Comment by GABRIELCMSANTOS** [comment]
ci:test:sf

**Comment by GABRIELCMSANTOS** [comment]
ci:test:sf

File: modules/apps/object/object-test/src/testFunctional/tests/PicklistExportImport.testcase [hunk 1]
```diff
@@ -313,9 +313,48 @@ definition {
...
+
+			IFrame.selectTopFrame();
+
+			Click(
+				key_text = "Continue",
+				locator1 = "Modal#MODAL_FOOTER_BUTTON");
+		}
+
+		task ("Then the existing Picklist is updated") {
+			AssertElementPresent(
```
Comment (epark18) [inline modules/apps/object/object-test/src/testFunctional/tests/PicklistExportImport.testcase:354]: Is this asserting the new picklist?

**Review by epark18 — COMMENTED** [review]
Hey @GABRIELCMSANTOS this looks good, just a couple questions.

File: modules/apps/object/object-test/src/testFunctional/macros/json/JSONObject.macro [hunk 2]
```diff
@@ -483,6 +483,34 @@ definition {
...
+		if (!(isSet(userPassword))) {
+			var userPassword = "test";
+		}
+
+		var curl = '''
+			${portalURL}/o/headless-admin-list-type/v1.0/list-type-definitions \
+				-H 'Content-Type: application/json' \
+				-u ${userEmailAddress}:${userPassword} \
+				-d '{
+					"externalReferenceCode": "${externalReferenceCode}",
```
Comment (timpak) [inline modules/apps/object/object-test/src/testFunctional/macros/json/JSONObject.macro:503]: What happens if we don't set an externalReferenceCode? @GABRIELCMSANTOS

File: modules/apps/object/object-test/src/testFunctional/tests/PicklistExportImport.testcase (hunk 1 above)
Comment (GABRIELCMSANTOS) [reply modules/apps/object/object-test/src/testFunctional/tests/PicklistExportImport.testcase:354]: Yes, it is.

File: modules/apps/object/object-test/src/testFunctional/macros/json/JSONObject.macro (hunk 2 above)
Comment (GABRIELCMSANTOS) [reply modules/apps/object/object-test/src/testFunctional/macros/json/JSONObject.macro:503]: On this API, I was trying to resume creating the Picklist, changing the ERC, and adding items, as required by the previous commentary. And the custom ERC is needed for the automation.

File: modules/apps/object/object-test/src/testFunctional/paths/Picklist.path [hunk 3]
```diff
@@ -55,6 +55,11 @@
 		<td>//button[@title='Add Item']</td>
 		<td></td>
 	</tr>
+	<tr>
+		<td>ASSERT_PICKLIST</td>
+		<td>//div[contains(@class,'table-list-title') and contains(.,'${key_picklistName}')]</td>
```
[the remark this answers is not in the document]
Comment (GABRIELCMSANTOS) [reply modules/apps/object/object-test/src/testFunctional/paths/Picklist.path:60]: If there is, I was not able to find it. It was a specific assert, since the element is a string inside a field.

File: modules/apps/object/object-test/src/testFunctional/macros/json/JSONObject.macro (hunk 2 above)
Comment (timpak) [reply modules/apps/object/object-test/src/testFunctional/macros/json/JSONObject.macro:503]: Ah let me rephrase the question. Can we use this macro to create the picklist without specifying the ERC? It's an unnecessary step for most tests so I wanted to know what happens if we call this macro without setting the ERC. If it works, that's fine. If it doesn't, then we should explore a better design so that we don't need to specify the ERC if we don't want to. Does that make sense?

Comment (GABRIELCMSANTOS) [reply modules/apps/object/object-test/src/testFunctional/macros/json/JSONObject.macro:503]: Yes, I got it. I will see if I can change the ERC after creating the picklist with the items.

Comment (epark18) [reply modules/apps/object/object-test/src/testFunctional/macros/json/JSONObject.macro:503]: @GABRIELCMSANTOS Sorry for the delay on this, but I meant that you could have created the Picklist Items with API instead of the Picklist itself. Not entirely sure if this test even needs Picklist Items in the first place to be honest. 

I think JSONObject.addPicklist allows a user to set an ERC so this new macro you created might not be necessary. Test looks good though.

Action Items:
1. Test to see if JSONObject.addPicklist will allow you to use a custom ERC. If not, creating the Picklist by itself via UI is okay. JSONObject.addPicklistWithCustomERC shouldn't be needed, so you do not need to submit that, but thank you for your work on it. Sorry for the confusion. 
2. If needed, change the Picklist Items steps to API.

Comment (GABRIELCMSANTOS) [reply modules/apps/object/object-test/src/testFunctional/macros/json/JSONObject.macro:503]: @epark18 No problem at all about the delay!
About the creation of the items, I did that so it could be visually verified that the picklist was changed with the update. But if it's not necessary, I can remove it. And I will see if I will be able to use SONObject.addPicklist to generate the ERC. Thanks for clarifying!

**Comment by GABRIELCMSANTOS** [comment]
@timpak Done! Now using a normal API and there are no more items. Is there anything that I am missing?

**Comment by epark18** [comment]
@GABRIELCMSANTOS this looks good, thank you for your work on this! Last thing is that the macro you created _addPicklistWithCustomERC_ isn't being used anymore so you can delete it from your PR. After that, you can forward. Thanks again!

**Comment by GABRIELCMSANTOS** [comment]
@epark18 Thanks! And about the removal of the unused API, I already deleted it, but somehow the changes did not appear. Can you check now if everything is ok?

**Comment by epark18** [comment]
ci:test:sf

**Comment by epark18** [comment]
Looks okay @GABRIELCMSANTOS. Just FYI, a commit like this https://github.com/liferay-bpm-qa/liferay-portal/pull/677/commits/8037c7478d6175924c08ea0d2944e9be56e72132 is not necessary. It's better to remove the commit entirely since that API macro that you are removing isn't in Liferay's code. It wouldn't make sense to have a commit that is removing code that doesn't exist in the code you are merging into. Other than that and the SF failure, this one is good.

**Comment by epark18** [comment]
ci:forward

**Comment by epark18** [comment]
Hey @GABRIELCMSANTOS your test didn't pass CI, can you take a look?

**Comment by GABRIELCMSANTOS** [comment]
@epark18 It was missing the dependencies folder. Should work now.

**Comment by epark18** [comment]
ci:forward

**Comment by epark18** [comment]
@GABRIELCMSANTOS remember to run your SF before pushing

**Comment by epark18** [comment]
ci:test:sf

**Comment by epark18** [comment]
@GABRIELCMSANTOS your SF is still failing. Also, can we rename the https://github.com/liferay-bpm-qa/liferay-portal/pull/677/commits/2c74a9fe0661b383192a7a293bffad273085468d message here since you aren't creating the macro with this commit.

**Comment by GABRIELCMSANTOS** [comment]
ci:test:sf

**Comment by GABRIELCMSANTOS** [comment]
ci:test:relevant

**Comment by GABRIELCMSANTOS** [comment]
@epark18 passed on relevant. Is there something more that I can do?

**Comment by epark18** [comment]
ci:forward
